src/ng/directive/input.js

```diff
@@ -1593,19 +1589,16 @@ function rangeInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     setInitialValueAndObserver(maxAttrType, 'max', maxChange);
   }
 
-  function setInitialValueAndObserver(attrType, attrName, changeFn) {
-    var initialAttrValue;
-    if (attrType === attrName) {
-      initialAttrValue = attr[attrName];
-      // Set the actual element attribute so that the browser can adjust the value based on
-      // the max value, which might be interpolated
-      element.attr(attrName, initialAttrValue);
+  function setInitialValueAndObserver(actualAttrName, htmlAttrName, changeFn) {
+    // e.g. max === max
+    if (actualAttrName === htmlAttrName) {
+      // interpolated attributes set the attribute value only after a digest, but we need the
+      // attribute value when the input is first rendered, so that the browser can adjust the
+      // input value based on the min/max value
+      element.attr(htmlAttrName, attr[htmlAttrName]);
```

Comment:
Yes, you are correct. If you put it like this, it indeed doesn't sound very useful. It's all because ngMin/Max et. al. doesn't set the actual attribute on the element. I assume it's done for a reason, but I haven't investigated.